internal/api/routes_test.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/broadinstitute/revere/internal/configuration"
+	"github.com/broadinstitute/revere/internal/version"
+	"github.com/gin-gonic/gin"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// Squelch Gin's normal logging output in favor of test logs
+var testConfig = configuration.Config{
+	Api: struct {
+		Port   int
+		Debug  bool
+		Silent bool
+	}{Debug: false, Silent: true},
+}
+
+// Alias the abstract fields needed to test a route
+type routeTest = struct {
+	name      string
+	reqMethod string
+	reqUrl    string
+	reqBody   io.Reader
+	wantCode  int
+	wantJson  interface{}
+}
+
+// Run an individual route test from within a larger test context `t`
+func runRouteTest(t *testing.T, rt routeTest) {
+	renderedWantJson, err := json.Marshal(rt.wantJson)
+	if err != nil {
+		t.Errorf("wantJson %v could not be rendered: %v", rt.wantJson, err)
+		return
+	}
+	router := NewRouter(&testConfig)
+	got := httptest.NewRecorder()
+	req, _ := http.NewRequest(rt.reqMethod, rt.reqUrl, rt.reqBody)
+	router.ServeHTTP(got, req)
+	if got.Code != rt.wantCode {
+		t.Errorf("%s %s -> code %d, want %d", rt.reqMethod, rt.reqUrl, got.Code, rt.wantCode)
+	}
+	if got.Body.String() != string(renderedWantJson) {
```

Comment:
I've been using the [diff method](https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff) from google's cmp package to do comparisons of arbitrary json or other objects in tests. It gives a nice human readable diff in the test output. Just a suggestion

Reply:
Great idea, to the point that I actually want to use that across Revere because I use a ton of `reflect.deepEquals`. I'll take that up next week I think